Manager/JobLauncher.php

```diff
@@ -0,0 +1,149 @@
+    /** @var string */
+    protected $environment;
+
+    /** @var array */
+    protected $config = [];
+
+    /**
+     * Constructor
+     *
+     * @param ObjectManager            $objectManager
+     * @param EventDispatcherInterface $eventDispatcher
+     * @param string                   $jobExecutionClass
+     * @param string                   $rootDir
+     * @param string                   $environment
+     */
+    public function __construct(
+        ObjectManager $objectManager,
+        EventDispatcherInterface $eventDispatcher,
+        $jobExecutionClass,
+        $rootDir,
+        $environment
+    ) {
+        $this->objectManager      = $objectManager;
+        $this->eventDispatcher    = $eventDispatcher;
+        $this->jobExecutionClass  = $jobExecutionClass;
+        $this->rootDir            = $rootDir;
+        $this->environment        = $environment;
+    }
+
+    /**
+     * Launch a job with command
+     *
+     * @param JobInstance   $jobInstance
+     * @param UserInterface $user
+     * @param string        $rawConfiguration
+     *
+     * @return JobExecution
+     */
+    public function launch(JobInstance $jobInstance, UserInterface $user, $rawConfiguration = null)
+    {
+        $jobExecution = $this->create($jobInstance, $user);
+        $executionId  = $jobExecution->getId();
+        $pathFinder   = new PhpExecutableFinder();
+
+        if ($this->hasConfig('upload')) {
+            $rawConfiguration.= ' ' . addslashes(json_encode($jobInstance->getJob()->getConfiguration()));
+        }
+
+        $cmd = sprintf(
+            '%s %s/console akeneo:batch:job --env=%s %s %s %s %s >> %s/logs/batch_execute.log 2>&1',
+            $pathFinder->find(),
+            $this->rootDir,
+            $this->environment,
+            $this->hasConfig('email') ? sprintf('--email="%s"', $user->getEmail()) : '',
+            $jobInstance->getCode(),
+            $executionId,
+            !empty($rawConfiguration) ? sprintf('--config="%s"', $rawConfiguration) : '',
+            $this->rootDir
+        );
+
+        // Please note we do not use Symfony Process as it has some problem
+        // when executed from HTTP request that stop fast (race condition that makes
+        // the process cloning fail when the parent process, i.e. HTTP request, stops
+        // at the same time)
+        exec($cmd . ' &');
+
+        $this->eventDispatcher->dispatch(JobProfileEvents::POST_EXECUTE, new GenericEvent($jobInstance));
+
+        return $jobExecution;
+    }
+
+    /**
+     * Set config
+     *
+     * @param array $config
+     *
+     * @return JobLauncher
+     */
+    public function setConfig(array $config = [])
+    {
+        $this->config = $config;
+
+        return $this;
+    }
+
+    /**
+     * Has config
+     *
+     * @param string $config
+     *
+     * @return bool
+     */
+    public function hasConfig($config)
+    {
+        return isset($this->config[$config]) && true === $this->config[$config];
```

Comment:
So it means that $this->config is an array containing only true or false ?

Because on this test, $this->config[$config] must contains extactly true